cross-chain/solana/solana-tbtc-anchor/programs/solana-tbtc-anchor/src/lib.rs

```diff
@@ -0,0 +1,362 @@
+    )]
+    pub guardian_info: Account<'info, GuardianInfo>,
+    pub guardian: Signer<'info>,
+}
+
+#[derive(Accounts)]
+pub struct Unpause<'info> {
+    #[account(
+        mut,
+        constraint = tbtc.paused @ TbtcError::IsNotPaused,
+        has_one = creator @ TbtcError::IsNotCreator
+    )]
+    pub tbtc: Account<'info, Tbtc>,
+    pub creator: Signer<'info>,
+}
+
+#[derive(Accounts)]
+pub struct SetupMint<'info> {
+    // Use the correct token mint for the program.
+    #[account(
+        mut,
+        seeds = [b"tbtc-mint", tbtc.key().as_ref()],
+        bump = tbtc.token_bump,
+        mint::decimals = 9,
+        mint::authority = tbtc_mint,
+    )]
+    pub tbtc_mint: Account<'info, SplMint>,
+    pub tbtc: Account<'info, Tbtc>,
+
+    // Set up the recipient's token account.
+    #[account(
+        init,
+        payer = payer,
+        associated_token::mint = tbtc_mint,
+        associated_token::authority = recipient,
+    )]
+    pub recipient_account: Account<'info, TokenAccount>,
+    /// CHECK: the recipient doesn't need to sign the mint,
+    /// and it doesn't conform to any specific rules.
+    /// Validating the recipient is the minter's responsibility.
+    pub recipient: UncheckedAccount<'info>,
+    #[account(mut)]
+    pub payer: Signer<'info>,
+
+    pub token_program: Program<'info, SplToken>,
+    pub associated_token_program: Program<'info, AssociatedToken>,
+    pub system_program: Program<'info, System>
+}
+
+#[derive(Accounts)]
+pub struct Mint<'info> {
+    // Use the correct token mint for the program.
+    #[account(
+        mut,
+        seeds = [b"tbtc-mint", tbtc.key().as_ref()],
+        bump = tbtc.token_bump,
+        mint::decimals = 9,
+        mint::authority = tbtc_mint,
+    )]
+    pub tbtc_mint: Account<'info, SplMint>,
+
+    // Can not mint when paused.
+    #[account(
+        constraint = !tbtc.paused @ TbtcError::IsPaused
+    )]
+    pub tbtc: Account<'info, Tbtc>,
+
+    // Require the signing minter to match a valid minter info.
+    #[account(
+        has_one = minter,
+        seeds = [b"minter-info", tbtc.key().as_ref(), minter.key().as_ref()],
+        bump = minter_info.bump,
+    )]
+    pub minter_info: Account<'info, MinterInfo>,
+    pub minter: Signer<'info>,
+
+    // Use the associated token account for the recipient.
+    // This account needs to be initialised first.
+    #[account(
+        mut,
+        associated_token::mint = tbtc_mint,
+        associated_token::authority = recipient,
+    )]
+    pub recipient_account: Account<'info, TokenAccount>,
+    /// CHECK: the recipient doesn't need to sign the mint,
+    /// and it doesn't conform to any specific rules.
+    /// Validating the recipient is the minter's responsibility.
+    pub recipient: UncheckedAccount<'info>,
+
+    pub token_program: Program<'info, SplToken>,
+}
+
+impl Tbtc {
+    pub const MAXIMUM_SIZE: usize = 32 + 32 + 1 + 1 + 1 + 1;
+}
+
+#[error_code]
+pub enum TbtcError {
```

Comment:
I noticed that in unit tests we only validated `IsPaused` error. Can we try to trigger the other ones too?

Reply:
Ok, up to you, but usually, we don't leave commented-out code like [here](https://github.com/keep-network/tbtc-v2/blob/solana-anchor/cross-chain/solana/solana-tbtc-anchor/programs/solana-tbtc-anchor/src/lib.rs#L36) or [here](https://github.com/keep-network/tbtc-v2/blob/solana-anchor/cross-chain/solana/solana-tbtc-anchor/programs/solana-tbtc-anchor/src/lib.rs#L53). I'd vote to remove it along with errors that are commented out in enum if we can't hit it. I'd just add a comment that `ConstraintSeeds` is hit first and guards these functions from unauthorized callers.